walkthroughs/week-4-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -14,10 +14,96 @@
 
 package com.google.sps;
 
+import java.util.ArrayList; 
+import java.util.Arrays;
 import java.util.Collection;
+import java.util.Collections;   
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
 
 public final class FindMeetingQuery {
-  public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
-    throw new UnsupportedOperationException("TODO: Implement this method.");
-  }
+
+    /**
+    * Returns a collection of available TimeRange given a collection of Event objects and
+    * MeetingRequest. A TimeRange is available if the person in MeetingRequest are all available.
+    * We first do this by checking which TimeRange are the attendees not available in and 
+    * use sliding window to record the available TimeRange provided they satisfy the duration.
+    * @param    events  a collection of Event objects
+    * @param    request a MeetingRequest object
+    * @return           collection of available TimeRange  
+    */
+    public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+        ArrayList<TimeRange> overlappingTime = new ArrayList<TimeRange>();
+        ArrayList<TimeRange> mergedEventTime = new ArrayList<TimeRange>();
```

Comment:
Nitpick: No need to initialize mergedEventTime as you assigned mergedEventTime below.

Reply:
ok noted!